src/Sharpliner/AzureDevOps/Model/Step.cs

```diff
@@ -70,6 +70,13 @@ public abstract record Step
     [DisallowNull]
     public ConditionedDictionary Env { get; init; } = [];
 
+    /// <summary>
+    /// Number of retries if the task fails.
+    /// Default is 0
+    /// </summary>
+    [YamlMember(Order = 230)]
+    public Conditioned<string>? RetryCountOnTaskFailure { get; init; }
```

Comment:
@PaulieC1972 I was told that this won't work on any `step` and maybe we should only put this on `task` to `AzureDevOpsTask`

Reply:
Seems to match the documentation which is for `step.task`

Reply:
Yes, I thought that might be the case.  I'll look into this again, but I couldn't find a good place to add this in for task only

Reply:
But completely understand that this retry is task bound and should not be available for steps